app/controllers/v1/submission_entries_controller.rb

```diff
@@ -7,6 +7,8 @@ def create
 
     return head :no_content if SubmissionEntry.exists?(submission_id: entry.submission_id, source: entry.source)
 
+    SubmissionEntryDataExtractor.new(entry).perform
```

Comment:
I feel like a cleaner abstraction would a simple data transformation one that modified the `params` and set the `:total_value`, rather than passing in and mutating the entry. e.g. we could end with something like `SubmissionEntryAttributes.new(params).to_hash` inside `submission_entry_params` such that `entry.attributes = submission_entry_params` has the desired effect.